src/common/ContractInteractor.ts

```diff
@@ -634,18 +634,19 @@ export default class ContractInteractor {
       })
     })
   }
-  
-    async getTransactionReceipt(transactionHash: PrefixedHexString, 
-        retries: number = WAIT_FOR_RECEIPT_RETRIES, 
-        initialBackoff: number = WAIT_FOR_RECEIPT_INITIAL_BACKOFF ): Promise<TransactionReceipt> {
+
+  async getTransactionReceipt (transactionHash: PrefixedHexString,
```

Comment:
One thing here could be that the initialBackoff could be calculated, example, with the contract interactor you can get the latest block and then retrieve the next before that, with the timestamps you can calculate the estimated mining time, or actually that could be a process running in background (or run it only once at the start) to calculate that value and then you can query that value from there in this function.

Reply:
Another idea could be to just get the chain id and knowing that regtest is 1 block per second estimated and testnet and mainnet are 30 seconds and 1 minute you could just use that to estimate the time to the initial backoff